openstf/model/serializer.py

```diff
@@ -123,32 +133,47 @@ def load_model(
 
         Returns:
             OpenstfRegressor: Loaded model
+            ModelSpecificationDataClass: model specifications
         """
         try:
-            experiment_id = self.setup_mlflow(pid)
+            experiment_id = self.setup_mlflow(modelspecs.id)
             # return the latest run of the model, .iloc[0] because it returns a list with max_results number of runs
             latest_run = mlflow.search_runs(
                 experiment_id,
                 filter_string="attribute.status = 'FINISHED'",
                 max_results=1,
             ).iloc[0]
+
             loaded_model = mlflow.sklearn.load_model(
                 os.path.join(latest_run.artifact_uri, "model/")
             )
+
+            # get the parameters from the old model, we insert these later into the new model
+            # get the hyper parameters from the previous model
+            modelspecs.hyper_params = loaded_model.get_params()
+            # get used feature names else use all feature names
+            try:
+                modelspecs.feature_names = json.loads(
+                    latest_run["tags.feature_names"].replace("'", '"')
```

Comment:
Nice! If we can load the feature names from the latest run, then I think that we don't have to load feature names from openstf_dbc anymore